styles/main.css

```diff
@@ -130,33 +139,44 @@ header li {
 header nav {
     float: right;
     width: auto;
-    display: inline-block;
     margin: -4px 2em 0;
     padding: 0 2em;
     border: 1px rgba(0, 0, 0, 0.3);
-    border-bottom-style: solid;
+    border-bottom-style: dotted;
+    overflow: auto;
+}
+
+header nav ul {
+    float: right;
 }
 
 header nav li {
     float: left;
-    padding: 1em 0 0.5em;
-    margin: 0 1em;
 }
 
-header nav li:hover {
-    margin-top: 0px;
+header nav li a {
+    padding: 1em 0 0.5em;
+    margin: 0 1em;
     border-top-style: solid;
     border-width: 4px;
-    border-color: fuchsia;
+    border-color: transparent;
+    display: block;
+}
+header nav li a:hover {
+    border-color: fuchsia;   
 }
 
 .back-to-top {
-    font-size: 2.5em;
     opacity: 0.5;
     text-align: right;
-    margin-right: 0.5em;
+    margin-right: -0.5em;
 }
 
 .back-to-top:hover {
     opacity: 1;
+}
+
+.back-to-top a {
+    position: fixed;
+    bottom: 3.5em;
 }
```

Comment:
Missing rule to position the back-to-top link at the bottom of the browser window